src/visitors/hoist.js

```diff
@@ -177,9 +187,124 @@ module.exports = {
         '$' + asset.id + '$require_resolve$' + t.toIdentifier(args[0].value);
       path.replaceWith(t.identifier(name));
     }
+  },
+  ExportAllDeclaration(path, asset) {
+    if (BREAK_COMMON_JS) {
+      path.replaceWith(
+        t.identifier(
+          '$' +
+            asset.id +
+            '$expand_exports$' +
+            t.toIdentifier(path.node.source.value)
+        )
+      );
+    } else {
+      path.replaceWith(
+        EXPORT_ALL_TEMPLATE({
+          EXPORTS: t.identifier('$' + asset.id + '$exports'),
+          SOURCE: t.identifier(
+            '$' +
+              asset.id +
+              '$require$' +
+              t.toIdentifier(path.node.source.value)
+          )
+        })
+      );
+    }
+  },
+  ImportDeclaration(path, asset) {
+    let {source, specifiers} = path.node;
+
+    if (source && specifiers.length > 0) {
+      if (BREAK_COMMON_JS) {
+        path.replaceWith(
+          t.variableDeclaration(
```

Comment:
I think it should be possible to avoid these extra variables by just renaming the local references to point to the imported one, e.g.

```javascript
path.scope.rename(specifier.local, IMPORT_NAME_HERE);
```

Reply:
Better! Fixed